// @chakra-ui/react/dist/esm/hooks/use-list-collection.js
"use strict";
"use client";
import { useState } from 'react';
import { useCallbackRef } from './use-callback-ref.js';
import { createListCollection } from '@ark-ui/react/collection';

function useListCollection(props) {
  const { initialItems = [], filter, limit, ...collectionOptions } = props;
  const create = (items) => {
    return createListCollection({ ...collectionOptions, items });
  };
  const [collection, setCollectionImpl] = useState(
    () => create(limit != null ? initialItems.slice(0, limit) : initialItems)
  );
  const setCollection = useCallbackRef((collection2) => {
    setCollectionImpl(
      limit == null ? collection2 : collection2.copy(collection2.items.slice(0, limit))
    );
  });
  return {
    collection,
    filter: (inputValue) => {
      if (!filter) return;
      let filtered = create(initialItems).filter(
        (itemString) => filter(itemString, inputValue)
      );
      setCollection(filtered);
    },
    set: useCallbackRef((items) => {
      setCollection(create(items));
    }),
    reset: useCallbackRef(() => {
      setCollection(create(initialItems));
    }),
    clear: useCallbackRef(() => {
      setCollection(create([]));
    }),
    insert: useCallbackRef((index, ...items) => {
      setCollection(collection.insert(index, ...items));
    }),
    insertBefore: useCallbackRef((value, ...items) => {
      setCollection(collection.insertBefore(value, ...items));
    }),
    insertAfter: useCallbackRef((value, ...items) => {
      setCollection(collection.insertAfter(value, ...items));
    }),
    remove: useCallbackRef((...itemOrValues) => {
      setCollection(collection.remove(...itemOrValues));
    }),
    move: useCallbackRef((value, to) => {
      setCollection(collection.move(value, to));
    }),
    moveBefore: useCallbackRef((value, ...values) => {
      setCollection(collection.moveBefore(value, ...values));
    }),
    moveAfter: useCallbackRef((value, ...values) => {
      setCollection(collection.moveAfter(value, ...values));
    }),
    reorder: useCallbackRef((from, to) => {
      setCollection(collection.reorder(from, to));
    }),
    append: useCallbackRef((...items) => {
      setCollection(collection.append(...items));
    }),
    prepend: useCallbackRef((...items) => {
      setCollection(collection.prepend(...items));
    }),
    update: useCallbackRef((value, item) => {
      setCollection(collection.update(value, item));
    })
  };
}

export { useListCollection };
